### DISCUSSION TITLE
antd4的穿梭框怎么实现一个左右侧都是table，左侧table的数据源来自接口，右侧的数据源搜索来自已经选中的？

### DISCUSSION THREAD

**Discussion by little077** [ant-design/ant-design#46782] [General]
### What problem does this feature solve?

我们项目的需求。

### What does the proposed API look like?

只要实现这个功能就行，快被折磨疯了

**Comment by pangao66** [comment]
https://[codesandbox.io/p/sandbox/objective-wing-6iqbx](https://codesandbox.io/p/sandbox/objective-wing-6iqbx)
![image](https://github.com/ant-design/ant-design/assets/20113031/74516573-7e7c-4d93-8e3c-226d6e95f268)
文档有介绍

**Reply by little077** [reply]
我知道，但是这仍然没有加上过滤的情况，我想要的是穿梭框里是table，然后支持分页，也可以支持搜索，左边搜索时，根据接口获取到最新的数据源。

**Reply by little077** [reply]
官网的demo里获取异步数据也没有包含search的情况，最后我用render和filteroption实现了前端过滤，功能是实现了，没有按照我想要的实现方式来。这里我说的table其实是不是table无所谓，完全可以自定义render来实现展示功能，逻辑问题就是异步搜索，只render={item => `${item.title}-${item.description}`这样写，然后去异步获取数据源是可以实现的，但是需要后端配合把targetkey传过去，除此之外，我目前不知道怎么实现了
